doc/tutorials/chapter_5/00_basic_modules.md

```diff
@@ -0,0 +1,226 @@
+
+    ```dart
+    class ExampleModule extends Module {
+      ExampleModule(Logic a) {
+        a = addInput('a', input);
+        final b = addOutput('b');
+        
+        // Your Logic must use the Logic from addInput/inputs
+        b <= a;
+      }
+    }
+    ```
+
+2. Any logic outside of a Module must consume the signals only via outputs (from the output or addOutput methods) of the Module.
+
+    ```dart
+    class ExampleModule extends Module {
+      ExampleModule(Logic a) {
+        a = addInput('a', input);
+        final b = addOutput('b');
+        
+        // Your Logic must use the Logic from addInput/inputs
+        b <= a;
+      }
+      // getter (Create a getter function to expose your output)
+      Logic get b => output('b');
+    }
+    ```
+
+3. Logic must be defined *before* the call to `super.build()`, which always must be called **at the end of the `build()` method** if it is overidden.
+
+The `Module` base class has an optional String argument 'name' which is an instance name.
+
+## First Module (One input, One output, Simple Logic)
+
+Let's take an example of how to create a simple ROHD module. The example below shows a simple module created with one `input` and one `output`. Notice that `addInput()` and `addOutput()` are used, as mentioned previously, to register input and output ports. Another thing to note is that the logic of the module (i.e., output <= input) is included inside the constructor so that the `.build()` instruction will pick up the logic during the execution process.
+
+```dart
+import 'package:rohd/rohd.dart';
+import 'package:test/test.dart';
+
+class SimpleModule extends Module {
+    // constructor
+    SimpleModule(Logic input) {
+        // register input port
+        // add inputs in the constructor, passing in the Logic it is connected to
+        // it's a good idea to re-set the input parameters so you don't accidentally use the wrong one
+        input = addInput('input_1', input);
+
+        // register output port
+        // add outputs in the constructor as well
+        // you can capture the output variable to a local variable for use
+        var output = addOutput('out');
+
+        // now you can define your logic
+        // this example is just a passthrough from 'input' to 'output'
+        output <= input;
+  }
+}
+
+void main() async {
+  final input = Const(1);
+  final simModule = SimpleModule(input);
+  await simModule.build();
+
+  test('should return input value',
+      () => expect(simModule.signals.first.value.toInt(), equals(1)));
+}
+```
+
+Do note that the `build()` method returns a `Future<void>`, not just `void`. This is because the `build()` method is permitted to consume real wallclock time in some cases, for example for setting up cosimulation with another simulator. If you expect your build to consume wallclock time, make sure the Simulator is aware it needs to wait before proceeding.
```

Comment:
The `build` should always be `await`ed, even if there's nothing known that will cause it to delay.  Otherwise dart will continue executing other things instead of `build`ing first.